src/mongocrypt-ctx.c

```diff
@@ -901,6 +901,18 @@ _mongocrypt_ctx_init (mongocrypt_ctx_t *ctx,
       }
    }
 
+   /* Check that the kms provider set and the datakey kms provider match. */
+   if (ctx->opts.kek.kms_provider) {
```

Comment:
Suggest moving this if block below the `OPT_PROHIBITED` check on L916.

The `kek` option, set through `mongocrypt_ctx_setopt_key_encryption_key` is only applicable to some contexts. If the `kek` option is set on an incorrect context, it would be a bug in the driver implementation. The "master key prohibited" error may be more helpful.